### PR TITLE
8259522: Apply java.io.Serial annotations in java.desktop

### PR DESCRIPTION
**Pull request by mrserb**
Please review the application of @java.io.Serial annotation (JDK-8202385) to types in the desktop module to enable stricter compile-time checking of serialization-related declarations.

This annotation can be applied to these methods in the module:

    private void writeObject(java.io.ObjectOutputStream stream) throws IOException
    private void readObject(java.io.ObjectInputStream stream) throws IOException, ClassNotFoundException
    private void readObjectNoData() throws ObjectStreamException
    ANY-ACCESS-MODIFIER Object writeReplace() throws ObjectStreamException
    ANY-ACCESS-MODIFIER Object readResolve() throws ObjectStreamException
    private static final ObjectStreamField[] serialPersistentFields
    private static final long serialVersionUID

Notes:
  - I have tried to update the comments for serialVersionUID as accurately as possible, but mostly based on the source code history and bugs in JBS where that field was added
  - Some of the readObject/writeObject methods in the javax.swing package does not have a spec, because this package and some others are excluded from the serialization specification.

A similar fix was implemented for java.base module as well:
http://mail.openjdk.java.net/pipermail/core-libs-dev/2019-August/062046.html

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8259522](https://bugs.openjdk.java.net/browse/JDK-8259522): Apply java.io.Serial annotations in java.desktop

### Reviewers
 * [Alexey Ivanov](https://openjdk.java.net/census#aivanov) (@aivanov-jdk - **Reviewer**) ⚠️ Review applies to 2e50a258d4a340f806d193369a6f45104a49f1b2
 * [Prasanta Sadhukhan](https://openjdk.java.net/census#psadhukhan) (@prsadhuk - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/2020/head:pull/2020`
`$ git checkout pull/2020`

### CODE REVIEW

**Review by aivanov-jdk — APPROVED** [review]

File: src/java.desktop/share/classes/com/sun/media/sound/InvalidDataException.java [hunk 1]
```diff
@@ -34,6 +35,10 @@
  */
 public class InvalidDataException extends IOException {
 
+    /**
+     * Use serialVersionUID from JDK 1.7 for interoperability.
+     */
+    @Serial
     private static final long serialVersionUID = 1L;
```
[the remark this answers is not in the document]
Comment (mrserb) [reply src/java.desktop/share/classes/com/sun/media/sound/InvalidDataException.java:42]: I think it is "quite" unique, the "1L" is used from the beginning. It is just different from the value which could be generated by the "serialver" but still should work fine.

[the remark this answers is not in the document]
Comment (aivanov-jdk) [reply src/java.desktop/share/classes/com/sun/media/sound/InvalidDataException.java:42]: Okay. There are several classes which extend this one, all of them use the same 1L value which makes it not "quite" unique.

Changing the values at this time presents more risks than benefits.

**Review by prsadhuk — COMMENTED** [review]
Why do we add serialVersionUID in some classes like DefaultMutableTreeNode.java but not in other swing classes?
Also, if this change is for stricter compile-time checking, shouldn't we remove @SuppressWarnings("serial") check?

**Comment by mrserb** [comment]
> Why do we add serialVersionUID in some classes like DefaultMutableTreeNode.java but not in other swing classes?

Most Swing classes are marked by the specific "Warning" that "Same-version serialization only" is supported. (I think such a warning is missed in a few classes). So generally the serialVersionUID field is not needed in such classes, but if present this provides a small benefit -> this UID is not generated at runtime.

For example, the DefaultMutableTreeNode was updated by the JDK-5017904 fix, which was unrelated to serialization but was targeted for the performance issue.

> Also, if this change is for stricter compile-time checking, shouldn't we remove @SuppressWarnings("serial") check?

At some point, we probably can remove it but will need to fix all serialization warnings which were disabled.

**Review by prsadhuk — APPROVED** [review]

**Pull request closed without merging**